### PR TITLE
add docs for DockerCli and NewDockerCli fixes #11593

### LINKED ISSUE #11593: Missing documentation in API client
**Problem**
 
Document `DockerCli` struct and `NewDockerCli` function in `api/client/cli.go`.
 
**Solution**
 
Write godoc-style comments which describe idea of DockerCli(each Cmd... method represents cli command) and NewDockerCli parameters. Examples of such comments can be found in golang standard library golang.org/pkg/.

### PR DESCRIPTION
**Pull request by buddhamagnet**
Fixes https://github.com/docker/docker/issues/11593

### CODE REVIEW

**Comment by buddhamagnet** [comment]
Hope this is good. I made the existing comments consistent (corrected grammar, added full stops), documented the DockerCli struct and the NewDockerCli function. Hope this helps!

File: api/client/cli.go [hunk 1]
```diff
@@ -20,25 +20,38 @@ import (
...
+	// configFile holds the configuration file (instance of registry.ConfigFile).
 	configFile *registry.ConfigFile
-	in         io.ReadCloser
-	out        io.Writer
-	err        io.Writer
-	keyFile    string
-	tlsConfig  *tls.Config
-	scheme     string
-	// inFd holds file descriptor of the client's STDIN, if it's a valid file
+	// in holds the instance of io.ReadCloser.
```
Comment (ahmetalpbalkan) [inline api/client/cli.go]: `in holds the input stream ...` (let's be consistent with out/err)

Comment (buddhamagnet) [inline api/client/cli.go]: Done @ahmetalpbalkan.

File: api/client/cli.go [hunk 2]
```diff
@@ -20,25 +20,38 @@ import (
 	"github.com/docker/docker/registry"
 )
 
+// The DockerCli struct represents the docker command line client.
```
Comment (ahmetalpbalkan) [inline api/client/cli.go]: `The` unnecessary. Golang comments must start with type/function name:

    DockerClu represents ...

Comment (LK4D4) [inline api/client/cli.go]: Should be like `DockerCli ...`.

Comment (buddhamagnet) [inline api/client/cli.go]: Amended, thanks @abhirajbutala @LK4D4.

**Comment by ahmetalpbalkan** [comment]
LGTM

**Comment by buddhamagnet** [comment]
OK @ahmetalpbalkan tests are green what's next?

**Comment by LK4D4** [comment]
LGTM

**Comment by buddhamagnet** [comment]
@ahmetalpbalkan @LK4D4 thanks guys!

**Comment by LK4D4** [comment]
@buddhamagnet Thank you for your work.

**Pull request merged**